ViewGeneratorCore/tools/src/ViewGenerator.ts

```diff
@@ -280,14 +280,18 @@ class Generator {
             `    protected override string NativeObjectName => \"${this.propsInterfaceCoreName}\";\n` +
             `    protected override string ModuleName => \"${this.filename}\";\n` +
             `    protected override object CreateNativeObject() => new ${PropertiesClassName}(this);\n` +
-            `    protected override string[] Events => ${this.propsInterface ? `new string[] { ${this.propsInterface.functions.map(p => `"${p.name}"`).join(",")} }` : ``};\n` +
-            `    protected override ${keyValuePairType}[] PropertiesValues {\n` +
-            `        get { \n` +
-            `            return new ${keyValuePairType}[] {\n` +
-            `    ${this.propsInterface ? f(this.propsInterface.properties.map(p => `            new ${keyValuePairType}("${p.name}", ${toPascalCase(p.name)})`).join(",\n")) : `` }\n` +
-            `            };\n` +
-            `        }\n` +
-            `    }\n` +
+            `${this.propsInterface && this.propsInterface.functions.length > 0
+                ? `    protected override string[] Events => new string[] { ${this.propsInterface.functions.map(p => `"${p.name}"`).join(",")} };\n`
+                : ``}` +
+            `${this.propsInterface && this.propsInterface.properties.length > 0
+                ? `    protected override ${keyValuePairType}[] PropertiesValues {\n` +
+                `        get { \n` +
+                `            return new ${keyValuePairType}[] {\n` +
+                `    ${f(this.propsInterface.properties.map(p => `            new ${keyValuePairType}("${p.name}", ${toPascalCase(p.name)})`).join(",\n")) }\n` +
+                `            };\n` +
+                `        }\n` +
+                `    }\n`
+                : ``}` +
```

Comment:
its a bit hard to read ... you better separate the dump into functions so that the code becomes with better indentation